packages/ui/src/TextField.tsx

```diff
@@ -90,54 +87,36 @@ export const TextField = ({
   }
 
   const [focused, setFocused] = useState(false);
-  const [height, setHeight] = useState(propsHeight || 40);
-  const [showDate, setShowDate] = useState(false);
-
-  const borderColor = useMemo(() => {
-    if (disabled) {
-      return theme.border.activeNeutral;
-    } else if (errorText) {
-      return theme.border.error;
-    } else {
-      return focused ? theme.border.focus : theme.border.dark;
-    }
-  }, [
-    disabled,
-    errorText,
-    focused,
-    theme.border.activeNeutral,
-    theme.border.dark,
-    theme.border.error,
-    theme.border.focus,
-  ]);
+  const [height, setHeight] = useState(rows * 40);
 
-  const getHeight = useCallback(() => {
-    if (grow) {
-      return Math.max(40, height);
-    } else if (multiline) {
-      return height || "100%";
-    } else {
-      return 20;
-    }
-  }, [grow, height, multiline]);
-
-  const defaultTextInputStyles = useMemo(() => {
-    const defaultStyles: StyleProp<TextStyleWithOutline> = {
-      flex: 1,
-      width: "100%",
-      height: getHeight(),
-      color: theme.text.primary,
-      fontFamily: theme.font.primary,
-      fontSize: 16,
-      paddingVertical: 0,
-    };
+  let borderColor = focused ? theme.border.focus : theme.border.dark;
```

Comment:
Hmm, maybe it makes sense since we'll re-render or each character, reverting. My thought was none of the stuff here is complex calculations, so it's possible the overhead of useMemo/useCallback negates the gains here (and at the expense of more complex/error prone code if we mess up the dependencies).